contracts/extensions/MultisigPermissions.sol

```diff
@@ -0,0 +1,551 @@
+
+  // Overrides
+
+  /// @notice Returns the identifier of the extension
+  /// @return _identifier The extension's identifier
+  function identifier() public pure override returns (bytes32 _identifier) {
+    return keccak256("MultisigPermissions");
+  }
+
+  /// @notice Returns the version of the extension
+  /// @return _version The extension's version number
+  function version() public pure override returns (uint256 _version) {
+    return 1;
+  }
+
+  /// @notice Configures the extension
+  /// @param _colony The colony in which the extension holds permissions
+  function install(address _colony) public override auth {
+    super.install(_colony);
+    colonyNetwork = IColonyNetwork(colony.getColonyNetwork());
+  }
+
+  function getCapabilityRoles(bytes4 _sig) public pure override returns (bytes32) {
+    if (_sig == SET_USER_ROLES) {
+      return ONLY_ROOT_ROLE_MASK | ONLY_ARCHITECTURE_ROLE_MASK;
+    }
+    return bytes32(0);
+  }
+
+  // Public
+
+  modifier motionExists(uint256 _motionId) {
+    require(_motionId > 0 && _motionId <= motionCount, "multisig-motion-nonexistent");
+    _;
+  }
+
+  modifier notExecuted(uint256 _motionId) {
+    require(!motions[_motionId].executed, "multisig-motion-already-executed");
+    _;
+  }
+
+  modifier notRejected(uint256 _motionId) {
+    require(!motions[_motionId].rejected, "multisig-motion-already-rejected");
+    _;
+  }
+
+  modifier onlyCoreRoot() {
+    require(
+      colony.hasUserRole(msgSender(), 1, ColonyDataTypes.ColonyRole.Root),
+      "multisig-permissions-not-core-root"
+    );
+    _;
+  }
+
+  function setGlobalThreshold(uint256 _globalThreshold) public onlyCoreRoot {
+    globalThreshold = _globalThreshold;
+
+    emit GlobalThresholdSet(_globalThreshold);
+  }
+
+  function setDomainSkillThreshold(uint256 _domainSkillId, uint256 _threshold) public onlyCoreRoot {
+    domainSkillThreshold[_domainSkillId] = _threshold;
+
+    emit DomainSkillThresholdSet(_domainSkillId, _threshold);
+  }
+
+  function createMotion(
+    uint256 _permissionDomainId,
+    uint256 _childSkillIndex,
+    address[] memory _targets,
+    bytes[] memory _data
+  ) public notDeprecated {
+    require(_data.length >= 1, "colony-multisig-invalid-motion");
+    require(_targets.length == _data.length, "colony-multisig-invalid-motion");
+
+    motionCount += 1;
+    Motion storage motion = motions[motionCount];
+    motion.targets = _targets;
+    motion.data = _data;
+    motion.creator = msgSender();
+    motion.creationTimestamp = block.timestamp;
+
+    for (uint256 i = 0; i < _data.length; i += 1) {
+      require(_targets[i] != address(colony), "multisig-passed-target-cannot-be-base-colony");
+
+      ActionSummary memory actionSummary = getActionSummary(
+        address(colonyNetwork),
+        address(colony),
+        _data[i],
+        _targets[i]
+      );
+
+      require(actionSummary.domainSkillId < type(uint256).max, "colony-multisig-invalid-motion");
```

Comment:
I assume we made an explicit choice to handle these errors in the caller, but I'm wondering if that was the right choice / if we can move this back into `getActionSummary`.

Reply:
I suspect it's because we wanted to reserve the right not to error and to return instead, but I can't remember exactly.

Reply:
Music to my ears! I've given it a go, but a fair few tests had to change (and I think this revealed some tests were incorrect, but would like input on that front). I've put these changes in [their own commit](https://github.com/JoinColony/colonyNetwork/pull/1144/commits/0b350709fe9da6c1b8a15890498e0a37d1de550e) so they're easier to look at standalone.